src/systems/filecoin_markets/storage_market/storage_market_actor_code.go

```diff
@@ -187,12 +188,24 @@ func (a *StorageMarketActorCode_I) ActivateDeals(rt Runtime, dealIDs []deal.Deal
 		dealP := publishedDeal.Deal().Proposal()
 		st._assertDealNotYetExpired(rt, dealP)
 
-		onchainDeal := st._activateDeal(rt, publishedDeal)
-		ret = append(ret, onchainDeal)
+		dealEnd := dealP.EndEpoch()
+
+		if dealEnd > lastDealExpiration {
+			lastDealExpiration = dealEnd
+		}
+
+		dealPs = append(dealPs, dealP)
+		st._activateDeal(rt, publishedDeal)
 	}
 
+	st._assertEpochEqual(rt, lastDealExpiration, sectorExpiration)
```

Comment:
Should be less-than-or-equal?

Reply:
currently, we expect sector Activation to be when ProveCommit happens and sector Expiration to be when the last deal expires, hence the equality check.